### PR TITLE
Upgraded console logger to log the proper levels. Debug, info, warn, error supported. ✨

### PR DESCRIPTION
**Pull request by Scalahansolo**
Added basic tests to make these are covered explicitly.

## Motivation

Making this change to fix https://github.com/DataDog/browser-sdk/issues/1329. 

This will make the data dog logger much easier to use as both a production logger as well as a development logger. Preserving the level that is passed to the logger in the console will now match people's expectations on where logs will land.

## Changes

Any logs passed to the console logger will now respect the appropriate level.

When using the following methods...

`datadogLogs.getLogger('client')?.debug("SEAN DD ERROR!")`
ScreenShot 04-16 at 08 58

`datadogLogs.getLogger('client')?.info("SEAN DD ERROR!")`
ScreenShot 04-16 at 08 59

`datadogLogs.getLogger('client')?.warn("SEAN DD ERROR!")`
ScreenShot 04-16 at 08 59

`datadogLogs.getLogger('client')?.error("SEAN DD ERROR!")`
ScreenShot 04-16 at 08 59

## Testing

- [ ] Local
- [ ] Staging
- [x] Unit
- [ ] End to end

---

I have gone over the [contributing](https://github.com/DataDog/browser-sdk/blob/main/CONTRIBUTING.md) documentation.

### CODE REVIEW

**Comment by Scalahansolo** [comment]
@BenoitZugmeyer Figured I would ping you here considering you had commented on the related issue about this suggestion. I think I covered all the bases here.

**Comment by Scalahansolo** [comment]
Since I took the above screenshots. I removed the status name from the message itself and it's implied by the level it's logged at.

**Comment by BenoitZugmeyer** [comment]
Thank you very much for your contribution! We have a rather big [refactoring of the Logs SDK code pending](https://github.com/DataDog/browser-sdk/pull/1497), and your PR will conflict with those changes. The refactoring will probably be merged next week, and we'll update and merge your PR after that.

**Comment by Scalahansolo** [comment]
These changes look good! If there is anything I can do to help get this over the line, lemme know!

File: packages/logs/src/domain/logsCollection/logger/loggerCollection.ts [hunk 1]
```diff
@@ -51,3 +57,10 @@ export function isAuthorized(status: StatusType, handlerType: HandlerType, logge
     STATUS_PRIORITIES[status] >= STATUS_PRIORITIES[logger.getLevel()] && includes(sanitizedHandlerType, handlerType)
   )
 }
+
+function sendToConsole(logsMessage: LogsMessage, logger: Logger) {
+  if (isAuthorized(logsMessage.status, HandlerType.console, logger)) {
+    const api = objectHasValue(ConsoleApiName, logsMessage.status) ? logsMessage.status : ConsoleApiName.log
+    display[api](logsMessage.message, combine(logger.getContext(), logsMessage.context))
+  }
+}
```
Comment (bcaudan) [inline packages/logs/src/domain/logsCollection/logger/loggerCollection.ts:66]: **💬 suggestion:** `sendToConsole` naming seems a bit odd now.
What about inlining it and maybe extract a `display(status, message, context)` to `display.ts` instead?
‏

**Review by bcaudan — APPROVED** [review]

**Comment by Scalahansolo** [comment]
Just wanted to follow up here? Is there anything left needing to be done for this PR?

**Review by amortemousque — APPROVED** [review]

**Comment by BenoitZugmeyer** [comment]
@Scalahansolo we finished the review process, I'll merge your PR then release it next week. Thank you for your patience!

**Pull request merged**